dpe/src/commands/sign.rs

```diff
@@ -54,28 +56,41 @@ impl<C: Crypto> CommandExecution<C> for SignCmd {
         let digest = Digest::new(&self.digest, DPE_PROFILE.alg_len())
             .map_err(|_| DpeErrorCode::InternalError)?;
         let algs = DPE_PROFILE.alg_len();
-        let priv_key = if self.uses_nd_derivation() {
-            dpe.contexts[idx].cached_priv_key.take().unwrap_or_else({
-                || {
-                    C::derive_ecdsa_key(
-                        algs,
-                        &dpe.derive_cdi(idx, true).unwrap(),
-                        &self.label,
-                        b"ECC",
-                    )
-                }
-            })
-        } else {
+
+        let EcdsaSig { r, s } = if self.uses_symmetric() {
```

Comment:
I think this part is still a bit confusing. Might be better to just diverge these paths instead of trying to return the same type from each branch.

Reply:
If we wanted to make this more clear, I think we have two options with tuples:

1. Use a tuple or `CryptoBuf`:

```Rust
        let (r, s) = if !self.uses_symmetric() {
            let EcdsaSig { r, s } = self.asymmetric_sign(dpe, idx, &digest)?;
            (r, s)
        } else {
            let r = self.symmetric_sign(dpe, idx, &digest)?;
            (r, CryptoBuf::default(DPE_PROFILE.alg_len()))
        };
```

2. use a tuple of `CryptoBuf` and `Option<CryptoBuf>`:

```Rust
        let (r, s) = if !self.uses_symmetric() {
            let EcdsaSig { r, s } = self.asymmetric_sign(dpe, idx, &digest)?;
            (r, Some(s))
        } else {
            let r = self.symmetric_sign(dpe, idx, &digest)?;
            (r, None)
        };

        dpe.roll_onetime_use_handle(idx)?;

        Ok(Response::Sign(SignResp {
            new_context_handle: dpe.contexts[idx].handle,
            sig_r_or_hmac: r.bytes().try_into().unwrap(),
            sig_s: s
                .unwrap_or_else(|| CryptoBuf::default(DPE_PROFILE.alg_len()))
                .bytes()
                .try_into()
                .unwrap(),
        }))
```